src/java/io/bazel/rulesscala/scalac/ScalacWorker.java

```diff
@@ -228,7 +232,35 @@ private static void compileScalaSources(CompileOptions ops, String[] scalaSource
     String[] compilerArgs =
         merge(ops.scalaOpts, ops.pluginArgs, constParams, pluginParams, scalaSources);
 
-    MainClass comp = new MainClass();
+    MainClass comp = new MainClass() {
+      private Global compiler;
+
+      @Override
+      public Global newCompiler() {
+        if (compiler == null) {
```

Comment:
basic question and I'm not a compiler dev but I seem to remember that the scala compiler has multi threaded parts.
Are we sure that `newCompiler` gets called in a thread safe way?

Reply:
From what I found, it seems to be. It's called [here](https://github.com/scala/scala/blob/3987e83b7cf0af7c8dcb17a08e8a040ccbe9a5e5/src/compiler/scala/tools/nsc/Driver.scala#L53)

Reply:
What in that link adds thread safety? Didn't see it

Reply:
The `process` method seems to be the only one to call the `newCompiler` and that is called directly from the Worker itself, so I believe that for each worker this call should be single-threaded. But perhaps some tests might bring that evidence to the table more easily 😃